.github/workflows/build-dependencies.yml

```diff
@@ -32,13 +32,14 @@ jobs:
     steps:
 
       - name: Checkout code into the directory
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Setup Go environment explicitly
-        uses: actions/setup-go@v3
+        uses: actions/setup-go@v5
         with:
-          go-version: "1.21"
+          go-version-file: 'go.mod'
```

Comment:
info: This is done so we now don't have to manually update the version every time we bump go version. This will detect it from the mod file automatically.